### ISSUE TITLE
Plugins and multiple users or new data structure

### ISSUE THREAD

**Issue by mattpass** [icecoder/ICEcoder#863]
If you install a plugin as user A, it's shown as installed. If you then login as user B, it's not shown as installed.

We should show the install button for user B and if they go to install it, simply set the config setting to installed and don't bother getting the zip. If they uninstall it should just change the config setting back to uninstalled.

A real drawback here is it's not really installing or uninstalling as expected. This also doesn't handle updates to the plugin very well.

Could look at versioning plugins here, or install/uninstall plugins on a per user basis.

This could be a better solution, as the `data` dir could truly contain all data created during ICEcoder use. For that reason we may want to consider moving the `tmp` dir into data and have the potential structure:

```
data
- user-a
  - backups
  - config
  - plugins
  - tmp
- user-b
  - backups
  - config
  - plugins
  - tmp
```

**Comment by cedricgeffroy** [comment]
For security reasons you might want to inform a user which plugin has been installed by another user. 
Let say a user knows a vulnerability and uninstalls a plugin, it might be important to inform him that this plugin is still installed by another user?

**Comment by mattpass** [comment]
A couple of issues with that idea - we don't know the uninstalled for a security reason, plus I'm hoping to further isolate users from each other in future (a common request) so informing them what other users have done seems concerning.

May need to think about an improved solution.

**Comment by cedricgeffroy** [comment]
Yes agreed. 
Maybe an admin user that has some kind of dashboard that lists plugins installed by each user?

**Comment by mattpass** [comment]
Yep, definitely. Admin users have been a common request and if I move backup, config & plugin data to a per user setup under the data dir (eg /data/user-aaa/plugins) it'd move towards stats & details per user.
